src/main/java/com/books/controllers/UserController.java

```diff
@@ -35,4 +42,22 @@ public ResponseEntity register(@RequestBody User user){
         return new ResponseEntity<>(HttpStatus.CREATED);
     }
 
+    /**
+     * Retrieves all books liked by the user.
+     * @param principal current logged in user.
+     * @param pageNum pagination number.
+     * @param entriesPerPage entries per page for pagination.
+     * @return all books liked by the user.
+     */
+    @GetMapping("/myLikes")
+    public ResponseEntity<List<Book>> getUserLikedBooks(Principal principal,
+                                                        @RequestParam(required = false, defaultValue = "0") int pageNum,
+                                                        @RequestParam(required = false, defaultValue = "20") int entriesPerPage){
```

Comment:
can you read the defult value for entries per page from properties file?

Reply:
Done